ui/src/lib/formikhelpers.ts

```diff
@@ -129,6 +134,65 @@ export function yamlToExperiments(yamlObj: any): Experiment {
   delete spec.scheduler
   delete spec.duration
 
+  if (kind === 'TimeChaos' && spec.time_offset) {
+    spec.offset = spec.time_offset
+    delete spec.time_offset
+  }
+
+  if (kind === 'KernelChaos' && spec.fail_kern_request) {
+    spec.fail_kernel_req = spec.fail_kern_request
+    delete spec.fail_kern_request
+
+    spec.fail_kernel_req.callchain = spec.fail_kernel_req.callchain.map((frame: CallchainFrame) => {
+      if (!frame.parameters) {
+        frame.parameters = ''
+      }
+
+      if (!frame.predicate) {
+        frame.predicate = ''
+      }
+
+      return frame
+    })
+
+    spec.fail_kernel_req = {
+      ...defaultExperimentSchema.target.kernel_chaos.fail_kernel_req,
+      ...spec.fail_kernel_req,
+    }
+  }
+
+  if (kind === 'StressChaos') {
+    spec.stressors.cpu = {
+      ...defaultExperimentSchema.target.stress_chaos.stressors.cpu,
+      ...spec.stressors.cpu,
+    }
+
+    spec.stressors.memory = {
+      ...defaultExperimentSchema.target.stress_chaos.stressors.memory,
+      ...spec.stressors.memory,
+    }
+  }
```

Comment:
About stress chaos @cwen0 Maybe we should refactor the API and add actions such  as `cpu, memory, mixes` just like IO chaos.

These two configs can both set at the same time, or you can only set cpu and memory. You can find an example here 
https://github.com/pingcap/chaos-mesh/blob/master/examples/burn-cpu.yaml

![Screenshot from 2020-07-08 15-58-06](https://user-images.githubusercontent.com/25150124/86964400-fec46c80-c133-11ea-972a-a4d1081b44e2.png)

Reply:
> About stress chaos @cwen0 Maybe we should refactor the API and add actions such as `cpu, memory, mixes` just like IO chaos.
> 
> These two configs can both set at the same time, or you can only set cpu and memory. You can find an example here
> https://github.com/pingcap/chaos-mesh/blob/master/examples/burn-cpu.yaml @g1eny0ung. In the current page, I have to set both CPU and memory stress
> 
> ![Screenshot from 2020-07-08 15-58-06](https://user-images.githubusercontent.com/25150124/86964400-fec46c80-c133-11ea-972a-a4d1081b44e2.png)

I agree with this. We can add the `action` field.

Reply:
I'm already giving StressChaos a select field to select the action. It's only used in the front end, the API will not be affected.

For API refactoring, we can open an issue to tracking it. It's not very urgent, we can finish it in the future.